### PR TITLE
Script to run Objective-C tests

### PR DESCRIPTION
**Pull request by jcanizales**
Usage: Call from anywhere. Enjoy.

Part of #2119.

Next steps: Find a suitable test server to run locally. Call this script from Jenkins.

### CODE REVIEW

File: src/objective-c/tests/run_tests.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+cd $(dirname $0)
+
+# TODO(jcanizales): Remove when Cocoapods issue #3823 is resolved.
+export COCOAPODS_DISABLE_DETERMINISTIC_UUIDS=YES
+pod install
+
+# xcodebuild is very verbose. We filter its output and tell Bash to fail if any
+# element of the pipe fails.
+set -o pipefail
```
Comment (murgatroid99) [inline src/objective-c/tests/run_tests.sh]: You should also `set -e` to make the script fail if a command fails.

File: src/objective-c/tests/run_tests.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# Return to current directory on error.
+trap 'cd - > /dev/null; exit 1' ERR
```
Comment (murgatroid99) [inline src/objective-c/tests/run_tests.sh]: This shouldn't be necessary. Unless Mac behaves differently, PWD doesn't propagate from child processes when they exit.

File: src/objective-c/tests/run_tests.sh [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+set -o pipefail
+XCODEBUILD_FILTER='(^===|^\*\*|\bfatal\b|\berror\b|\bwarning\b|\bfail)'
+xcodebuild \
+    -workspace Tests.xcworkspace \
+    -scheme AllTests \
+    -destination name="iPhone 6" \
+    test \
+    | egrep "$XCODEBUILD_FILTER" -
+
+cd - > /dev/null
```
Comment (murgatroid99) [inline src/objective-c/tests/run_tests.sh]: For the same reason as above, this shouldn't be necessary.

**Comment by jcanizales** [comment]
To have the cross-reference: This uses a workaround for https://github.com/CocoaPods/CocoaPods/issues/3823.

File: src/objective-c/tests/run_tests.sh (hunk 2 above)
Comment (jcanizales) [inline src/objective-c/tests/run_tests.sh]: Nice! Done.

File: src/objective-c/tests/run_tests.sh (hunk 3 above)
Comment (jcanizales) [inline src/objective-c/tests/run_tests.sh]: Done.

File: src/objective-c/tests/run_tests.sh (hunk 1 above)
Comment (jcanizales) [inline src/objective-c/tests/run_tests.sh]: `trap 'exit 1' ERR` did that, but as I don't need to `cd -`, `set -e` is easier. Done.

**Comment by murgatroid99** [comment]
Also, I'm not sure how to read the `xscheme` file, but we should make sure we're not running the tests that hit the grpc-testing backend. And if I remember right, some of the tests require a running local server, so we should either run that or skip those tests.

**Comment by jcanizales** [comment]
> we should make sure we're not running the tests that hit the grpc-testing backend

That's totally what this script does. Just what we did manually before, only now automated. I agree that this is _not_ what we should do in Jenkins. For that we'll have to separate the interop tests into a different scheme. Until then, the script simplifies our manual testing.

> some of the tests require a running local server, so we should either run that or skip those tests

The scheme is currently set to skip them, until we decide on a suitable server, adjust the tests accordingly (those tests were written for the route_guide server in grpc-common), and add the command to run the server.

We'll want a server that can do both cleartext and SSL, and #2429 fixed to test against the local SSL server.

**Comment by jcanizales** [comment]
Note: Can be simplified by #2540.

**Pull request merged**